cli/credential.go

```diff
@@ -79,6 +109,84 @@ func createCredential(c *cli.Context) {
 	createCredentialImpl(c.String, c.Bool, cbClient.Cloudbreak.V1credentials)
 }
 
+func modifyCredential(c *cli.Context) {
+	checkRequiredFlagsAndArguments(c)
+	defer utils.TimeTrack(time.Now(), "modify credential")
+
+	cbClient := NewCloudbreakHTTPClientFromContext(c)
+	modifyCredentialImpl(c.String, c.Bool, cbClient.Cloudbreak.V1credentials)
+}
+
+type modifyCredentialClient interface {
+	PutPrivateCredential(params *v1credentials.PutPrivateCredentialParams) (*v1credentials.PutPrivateCredentialOK, error)
+	PutPublicCredential(params *v1credentials.PutPublicCredentialParams) (*v1credentials.PutPublicCredentialOK, error)
+	GetPublicCredential(params *v1credentials.GetPublicCredentialParams) (*v1credentials.GetPublicCredentialOK, error)
+}
+
+func modifyCredentialImpl(stringFinder func(string) string, boolFinder func(string) bool, client modifyCredentialClient) *models_cloudbreak.CredentialResponse {
+	name := stringFinder(FlName.Name)
+	description := stringFinder(FlDescriptionOptional.Name)
+	provider := cloud.GetProvider()
+
+	credential := getCredential(name, client)
+	if *credential.CloudPlatform != *provider.GetName() {
+		utils.LogErrorAndExit(errors.New("cloud provider cannot be modified"))
+	}
+
+	log.Infof("[modifyCredentialImpl] original credential found name: %s id: %d", name, credential.ID)
+	credentialMap, err := provider.GetCredentialParameters(stringFinder, boolFinder)
+	if err != nil {
+		utils.LogErrorAndExit(err)
+	}
+
+	if len(description) == 0 {
+		origDesc := credential.Description
+		if origDesc != nil && len(*origDesc) > 0 {
```

Comment:
It's not necessary to check nil here, len(nil) is 0

Reply:
That's true only for strings, but wen you try to read value of a nil pointer, it will occure:
`panic: runtime error: invalid memory address or nil pointer dereference`

Reply:
Right, but in this case CredentialResponse.Description is a *string